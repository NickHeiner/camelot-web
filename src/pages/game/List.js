import React, {PureComponent} from 'react';
import { Button } from 'react-bootstrap';
import autobind from 'autobind-decorator';
import firebase from 'firebase';
import _ from 'lodash';
import { Link } from 'react-router';

class GameList extends PureComponent {
    constructor() {
        super();
        this.state = {count: null};
    }

    componentWillMount() {
        this.gamesRef = firebase.database().ref('games');
        this.gamesRef.on('value', this.onGamesUpdate);
    }
    
    componentWillUnmount() {
        this.gamesRef.off('value', this.onGamesUpdate);
    }

    @autobind
    onGamesUpdate(snapshot) {
        const val = snapshot.val();
        if (val) {
            this.setState({games: val});
        }
    }

    render() {
        return <div>
            <ul>
                {
                    _.map(this.state.games, ((game, key) => <li key={key}><Link to={`play/${key}`}>{game.host}</Link></li>))
                }
            </ul>
            <Button bsStyle="primary" onClick={this.createNewGame}>New</Button>
        </div>;
    }

    @autobind
    createNewGame() {
        this.gamesRef.push({
            host: this.props.params.currentUser.uid
        });
    }
}

export default GameList;
