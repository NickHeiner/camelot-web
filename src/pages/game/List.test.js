import {PresentationGameList} from './List';
import React from 'React';
import { shallow } from 'enzyme';

describe('GameList', () => {
    it('renders with no games', () => {
        expect(shallow(
            <PresentationGameList />
        ).html()).toMatchSnapshot();
    });
    it('renders with games', () => {
        const games = [
            {
                host: 'Game Host 1',
                key: 'game-key-1'
            },
            {
                host: 'Game Host 2',
                key: 'game-key-2'
            }
        ];
        expect(shallow(
            <PresentationGameList games={games} />
        ).html()).toMatchSnapshot();
    });
});
