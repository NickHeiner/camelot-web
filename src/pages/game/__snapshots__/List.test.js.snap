exports[`GameList renders with games 1`] = `"<div><ul><li><a>Game Host 1</a></li><li><a>Game Host 2</a></li></ul><button type=\"button\" class=\"btn btn-primary\">New</button></div>"`;

exports[`GameList renders with no games 1`] = `"<div><ul></ul><button type=\"button\" class=\"btn btn-primary\">New</button></div>"`;
